lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -120,12 +130,15 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   }
 
   std::uint32_t InstNum = m_StartInstruction;
-  std::map<llvm::StringRef, std::pair<int, int>> InstructionRangeByFunctionName;
+  llvm::StringMap<std::pair<int, int>> InstructionRangeByFunctionName;
```

Comment:
I don't think llvm::StringMap is guaranteed to be determinstic in order when iterating.

Reply:
Yea, but the problem that Adam is pointing out here is real. PIX may not care about the order, but we don't want the compiler generating output non-deterministically, and since you're iterating over the map and printing output.

StringMap technically is deterministic, but it is unordered, so the ordering may not make sense when observed. Additionally, the ordering is not guaranteed to be stable across LLVM versions so relying on it will impact our move to modern LLVM.

One possible solution I see would be to merge the loops by adding the `OSOverride` printing happen inside the loop over the `instrumentableFunctions`.

Reply:
Fixed by removing map, per Chris' suggestion.